pyomo/core/plugins/transform/scaling.py

```diff
@@ -87,6 +87,10 @@ def _get_float_scaling_factor(self, instance, component_data):
             scaling_factor = instance.scaling_factor[component_data]
         elif component_data.parent_component() in instance.scaling_factor:
             scaling_factor = instance.scaling_factor[component_data.parent_component()]
+        elif component_data in component_data.parent_block().scaling_factor:
+            scaling_factor = component_data.parent_block().scaling_factor[component_data]
+        elif component_data.parent_component() in component_data.parent_block().scaling_factor:
+            scaling_factor = component_data.parent_block().scaling_factor[component_data.parent_component()]
```

Comment:
This makes the assumption that the `parent_block` *has* a `scaling_factor` Suffix.  If it is not present, this will raise an AtributeError.  I think that a better solution may be to follow the pattern in GDP's logic for hunting fors BigM suffixes (start at the parent component and walk up each block in the hierarchy looking for the first Suffix that "matches".